azavea/pfb-network-connectivity
# Update Tilegarden node to 12.x

## Overview

Updates the Tilegarden node version to 12 locally, and updates all node dependencies.

This PR is intended to test what happens in the AWS environment when the update gets deployed to staging; if it works fine, this will close #801. I expect however that there will be AWS Lambda environment-specific errors with `node-mapnik`.


### Notes

I've manually updated the node runtime on AWS for the staging environment to 12 in preparation for merging this. I couldn't find anything in the deployment directory to set the runtime, so it appears to be a manual configuration. The Tilegarden manual deployment instructions look to be very involved, so I think the easiest way to test how this behaves in the AWS environment would be to let CI deploy to staging once this PR is merged in.

Given [this issue with node-mapnik](https://github.com/mapnik/node-mapnik/issues/937), I expect that there will be `node-mapnik` errors on AWS that do not occur locally. I think it may be helpful to confirm those errors exist before proceeding to implement a custom runtime (as described in the linked issue).


## Testing Instructions

 * Run `./scripts/update` to rebuild Tilegarden
 * Tiles should still serve locally without error


## Checklist


Connects #801
